src/Microsoft.Graph.Core/Requests/Middleware/Options/AuthenticationHandlerOption.cs

```diff
@@ -4,10 +4,12 @@
 
 namespace Microsoft.Graph
 {
+    using Microsoft.Kiota.Abstractions;
+
     /// <summary>
     /// The auth middleware option class
     /// </summary>
-    public class AuthenticationHandlerOption : IMiddlewareOption
+    public class AuthenticationHandlerOption : IRequestOption
```

Comment:
not needed anymore

Reply:
To avoid making the PR too large and easier to review(and still have it compile) My intention is to refactor all the auth code via #334. 
If it's okay you will see this is done there.